src/main/kotlin/com/criptext/mail/api/PeerEventsApiHandler.kt

```diff
@@ -27,17 +33,43 @@ interface PeerEventsApiHandler {
 
         override fun dispatchEvents(): Result<Unit, Exception> {
             val picks = queue.pick()
-            return queue.dispatchAndDequeue(picks)
+            val op = workOperation(picks)
+
+            val sessionExpired = HttpErrorHandlingHelper.didFailBecauseInvalidSession(op)
+
+            return if(sessionExpired)
+                newRetryWithNewSessionOperation(picks)
+            else
+                op
         }
 
         private fun ServerErrorException.isQueueableError(): Boolean{
             if(errorCode >= ServerErrorCodes.InternalServerError) return true
             when(errorCode){
-                ServerErrorCodes.Unauthorized,
+                ServerErrorCodes.DeviceRemoved,
                 ServerErrorCodes.Forbidden,
                 ServerErrorCodes.TooManyRequests -> return true
             }
             return false
         }
+
+        private fun workOperation(picks: List<PendingEvent>) : Result<Unit, Exception> =
+                queue.dispatchAndDequeue(picks)
+
+        private fun newRetryWithNewSessionOperation(picks: List<PendingEvent>)
+                : Result<Unit, Exception> {
+            val refreshOperation =  HttpErrorHandlingHelper.newRefreshSessionOperation(apiClient, activeAccount, storage, accountDao)
+                    .mapError(HttpErrorHandlingHelper.httpExceptionsToNetworkExceptions)
+            return when(refreshOperation){
+                is Result.Success -> {
+                    val account = ActiveAccount.loadFromStorage(storage)!!
+                    apiClient = PeerAPIClient(httpClient, account.jwt)
```

Comment:
is there a way to only set the new `jwt` instead of create a new `apiClient` object? I don't like the idea of making `apiClient` mutable